tree.go

```diff
@@ -22,33 +22,65 @@ type Tree interface {
 	Root() ([]byte, error)
 }
 
-var _ Tree = &DefaultTree{}
+// treeFns is a global map used for keeping track of registered tree constructors for JSON serialization
+// The keys of this map should be kebab cased. E.g. "default-tree"
```

Comment:
[Question] Could you elaborate on the potential consequences of not following this convention? If enforcing kebab-case for names is necessary, it would be beneficial to implement a name check during the registration of tree names i.e., in the `RegisterTree` function.

Reply:
i think this is just a recommendation for a goal of consistent tree naming convention. [ref](https://github.com/celestiaorg/rsmt2d/pull/277#discussion_r1425791568)